mesonbuild/modules/gnome.py

```diff
@@ -1055,47 +1056,92 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs: 'Y
                 raise MesonException('Yelp requires a list of sources')
         elif args[1]:
             mlog.warning('"gnome.yelp" ignores positional sources arguments when the "sources" keyword argument is set')
-        source_str = '@@'.join(sources)
+        sources_files = [mesonlib.File.from_source_file(state.environment.source_dir,
+                                                        os.path.join(state.subdir, 'C'),
+                                                        s) for s in sources]
 
         langs = kwargs['languages']
+        if not langs:
+            langs = read_linguas(state.subdir)
+
+        media = kwargs['media']
+        symlinks = kwargs['symlink_media']
+        targets: T.List['Target'] = []
+        potargets: T.List['build.RunTarget'] = []
+
+        install_dir = kwargs.get('install_dir', os.path.join(state.environment.get_datadir(), 'help'))
+        c_install_dir = os.path.join(install_dir, 'C', project_id)
+        c_data = build.Data(sources_files, c_install_dir, c_install_dir,
+                            mesonlib.FileMode(), state.subproject)
+        targets.append(c_data)
+
+        media_files = []
+        for m in media:
+            f = mesonlib.File.from_source_file(state.environment.source_dir,
+                                               os.path.join(state.subdir, 'C'), m)
+            media_files.append(f)
+            m_install_dir = os.path.join(c_install_dir, os.path.dirname(m))
+            m_data = build.Data(media_files, m_install_dir, m_install_dir,
```

Comment:
Why do we build a list of media_files and on each pass, create a new build.Data object with all media_files found so far?

Shouldn't we just use `build.Data(f, ...)` here?

Reply:
Yes, that's a bug, thanks! The list is still need for the case when a certain locale does not have media files and they have to be installed from the "C" locale.